### DISCUSSION TITLE
cmt_decode_prometheus_parser.h

### DISCUSSION THREAD

**Discussion by effendid** [fluent/fluent-bit#6003] [Q&A] [answered]
Hello Everyone,

I encounter below issue during build fluent bit on my VM (Oracle Linux 8). Any idea of this?
`/users/thirdparty/fluent-bit/lib/cmetrics/include/cmetrics/cmt_decode_prometheus.h:96:10: fatal error: cmt_decode_prometheus_parser.h: No such file or directory
 #include "cmt_decode_prometheus_parser.h"
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [plugins/in_prometheus_scrape/CMakeFiles/flb-plugin-in_prometheus_scrape.dir/build.make:76: plugins/in_prometheus_scrape/CMakeFiles/flb-plugin-in_prometheus_scrape.dir/prom_scrape.c.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:3923: plugins/in_prometheus_scrape/CMakeFiles/flb-plugin-in_prometheus_scrape.dir/all] Error 2
make: *** [Makefile:156: all] Error 2
`

Any idea?

Thanks and stay healthy

**Comment by effendid** [comment, marked as answer]
please ignore me, using source package is OK, but not with clone from git.
So, I think it was expected behavior.

**Comment by korpx-z** [comment]
I'm seeing this same error now while attempting to build via git clone (tag/v2.1.9)
Would you mind elaborating on how you got around this issue?

**Reply by patrick-stephens** [reply]
Does it work with the official build process?
`./packaging/build.sh -d centos/7` for example - replace `centos/7` with a distro you want, it all runs in a containerised build.

Those are just run via a standard git clone as well so curious.

How are you configuring cmake to do the build?

**Comment by matschaffer-roblox** [comment]
I ran into this on macos.

Looks like the issue was that I was using the OS-provided bison (2.3) & m4 (just a stub). Using a homebrew-provided versions got the build passing, I just had to make sure this was in place so it used the right tools.

```
export PATH="/opt/homebrew/opt/bison/bin:$PATH"
export PATH="/opt/homebrew/opt/m4/bin:$PATH"
```

**Reply by patrick-stephens** [reply]
Worth adding a note to the macOS build docs?

**Reply by matschaffer-roblox** [reply]
https://docs.fluentbit.io/manual/installation/macos#install-build-dependencies looks like it has it pretty well covered.

I read https://github.com/fluent/fluent-bit?tab=readme-ov-file#quick-start and stopped there and also didn't know about the Bison >=3 requirement until I saw https://docs.fluentbit.io/manual/installation/sources/build-and-install

Wonder if it might be better to remove the "Quick Start" from the readme and point people to https://docs.fluentbit.io/manual/installation/getting-started-with-fluent-bit#compile-from-source-linux-windows-freebsd-macos

**Reply by patrick-stephens** [reply]
Sure, suggest a PR with reasoning

**Reply by matschaffer-roblox** [reply]
Done! https://github.com/fluent/fluent-bit/pull/8990
